mitiq/lre/lre.py

```diff
@@ -0,0 +1,161 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+
+"""Extrapolation methods for Layerwise Richardson Extrapolation (LRE)"""
+
+from functools import wraps
+from typing import Any, Callable, Optional, Union
+
+import numpy as np
+from cirq import Circuit
+
+from mitiq import QuantumResult
+from mitiq.lre import (
+    multivariate_layer_scaling,
+    multivariate_richardson_coefficients,
+)
+from mitiq.zne.scaling import fold_gates_at_random
+
+
+def execute_with_lre(
+    input_circuit: Circuit,
+    executor: Callable[[Circuit], QuantumResult],
```

Comment:
```suggestion
    executor: Callable[[Circuit], float],
```
I think this has to be a float, right? And right now, this just works with `cirq.Circuit` objects, right?

Reply:
Yes, this only works for cirq circuits right now. So, your suggestion does make sense. I decided to make it `QuantumResult` as `make check-types` failed with the following:

```
Argument 2 to "execute_with_lre" has incompatible type "Callable[[Circuit], Union[float, MeasurementResult, ndarray[Any, Any]]]"; expected "Callable[[Circuit], float]"  [arg-type]
```

Reply:
It should work if you change all the instances of `QuantumResult` in this file to `float`.